cclient/iota_client_extended_api.c

```diff
@@ -235,3 +235,63 @@ retcode_t iota_client_get_account_data(iota_client_service_t const* const serv,
   get_balances_res_free(&blances_res);
   return ret_code;
 }
+
+retcode_t iota_client_find_transaction_objects(
+    iota_client_service_t const* const serv, hashes_t* addresses,
+    hashes_t const* const bundles, hashes_t const* const tags,
+    hashes_t const* const approvees, transaction_list_t* out_tx_objs) {
+  retcode_t ret_code = RC_OK;
+
+  log_debug(CCLIENT_EXTENDED_LOGGER_ID, "[%s:%d]\n", __func__, __LINE__);
+
+  find_transactions_req_t* find_tx_req = NULL;
+  find_transactions_res_t* find_tx_res = NULL;
+  get_trytes_req_t* get_trytes_req = get_trytes_req_new();
+  get_trytes_res_t* get_trytes_res = get_trytes_res_new();
+
+  if (!find_tx_req || !find_tx_res || !get_trytes_req || !get_trytes_res) {
+    ret_code = RC_CCLIENT_NULL_PTR;
+    goto done;
+  }
+
+  &find_tx_req->addresses = flex_hash_array_to_hash243_queue(
+      addresses) if (!(&find_tx_req->addresses)) {
+    goto done;
+  }
+  &find_tx_req->bundles =
+      flex_hash_array_to_hash243_queue(bundles) if (!(&find_tx_req->bundles)) {
+    goto done;
+  }
+  &find_tx_req->tags =
+      flex_hash_array_to_hash81_queue(tags) if (!(&find_tx_req->tags)) {
+    goto done;
+  }
+  &find_tx_req->approvees = flex_hash_array_to_hash243_queue(
+      approvees) if (!(&find_tx_req->approvees)) {
+    goto done;
+  }
+  ret_code = iota_client_find_transactions(serv, find_tx_req, find_tx_res);
+  if (ret_code) {
+    goto done;
+  }
+
+  get_trytes_req->hashes = find_tx_res->hashes;
+  if (!get_trytes_req->hashes) {
+    goto done;
+  }
+
+  ret_code = iota_client_get_trytes(sev, get_trytes_req, get_trytes_res);
+  if (ret_code) {
+    goto done;
+  }
+
+  out_tx_obj = hash243_queue_to_flex_hash_array(get_trytes_res->trytes);
```

Comment:
should return `transaction objects`, not `trytes`.

Reply:
sorry for the errors. I will fix them asap

Reply:
the transaction object I mentioned is `iota_transaction_t`.

Reply:
got it !
thank you for the information